llvm/test/Transforms/PGOProfile/global_variables_only.ll

```diff
@@ -0,0 +1,5 @@
+; RUN: opt < %s -passes=pgo-instr-gen -S | FileCheck %s --implicit-check-not='__llvm_profile_raw_version'
```

Comment:
While a module that has global variable definitions doesn't create PGO counters, it can provide other information in FDO profiles. 

For instance, dynamic type profiling collects the vtable names in raw FDO profiles (explained in https://discourse.llvm.org/t/rfc-dynamic-type-profiling-and-optimizations-in-llvm/74600).
The RFC is under development but https://github.com/llvm/llvm-project/pull/81442 should be the core change for more efficient indirect-call-promotion.

If the motivation of this change is to speed up instrumentation build (which makes sense), what do you think of a `cl::opt` that allows instrumentation of global variables?

Reply:
With all above said, I _think_ it's possible for a module to have global variables (say `file.cc` defines a list of constants, if `file.h` is not suitable), but not that common for a module to have vtable definitions only.

Reply:
@minglotus-6 Thank you for the comment. Maybe I've chosen wrong wording for the title and some description. The patch doesn't disable any instrumentation indeed, it just disables the variable `__llvm_profile_raw_version` generation when nothing has actually been instrumented in the module. The test just demonstrates the current behavior. When a solution to instrument global variables is landed, the `__llvm_profile_raw_version` variable will be generated and the test will ought to be rewritten. This will make some issues when already split thin-LTO modules is put under post-link instrumentation though, but this is unrelated to the PR. Actually my intent is to open a door for instrumentation during the post-link phase, when the modules have already been split. When the `__llvm_profile_raw_version` variable presents twice in the same even thought split module, it makes a problem with a duplicated symbol for the linker.

Reply:
thanks for the clarification! I was reading the description before. Now reading the code I saw this patch doesn't affect profile generation, so my question is not relevant anymore :)

Reply:
Thank you for the questions, they let me get that the commit header and description were inaccurate. I've made them (at least I believe) a bit more precise.